tony-core/src/main/java/com/linkedin/tony/TonyClient.java

```diff
@@ -678,53 +678,21 @@ protected ApplicationRpcClient getAMRpcClient() {
     return this.amRpcClient;
   }
 
-  /**
-   * Clean up temporary files.
-   */
-  private void cleanUp() {
-    try {
-      if (amRpcClient != null) {
-        amRpcClient.finishApplication();
-      }
-    } catch (IOException | YarnException e) {
-      LOG.error("Failed to finish application.", e);
-    } finally {
-      Utils.cleanupHDFSPath(hdfsConf, appResourcesPath);
-    }
-  }
-
-  public static TonyClient createClientInstance(String[] args, Configuration conf) throws ParseException {
-    TonyClient client;
-    client = new TonyClient(conf);
-    boolean sanityCheck = client.init(args);
-    if (!sanityCheck) {
-      LOG.fatal("Failed to init client.");
-      return null;
+  @Override
+  public void close() throws IOException, YarnException {
+    if (amRpcClient != null) {
+      amRpcClient.finishApplication();
     }
-    return client;
-  }
-
-  public static int start(String[] args) {
-    return start(args, new Configuration(false));
+    Utils.cleanupHDFSPath(hdfsConf, appResourcesPath);
```

Comment:
I don't think we can refactor it like this? if finishApplication throws exception then appResourcesPath will not be cleaned up

Reply:
That is right. Will update this.